pyproject.toml

```diff
@@ -99,14 +99,19 @@ env = { TEMPORAL_INTEGRATION_TEST = "1" }
 cmd = "pip uninstall temporalio -y"
 
 [tool.pytest.ini_options]
-addopts = "-p no:warnings"
 asyncio_mode = "auto"
 log_cli = true
 log_cli_level = "INFO"
 log_cli_format = "%(asctime)s [%(levelname)8s] %(message)s (%(filename)s:%(lineno)s)"
 testpaths = ["tests"]
 timeout = 600
 timeout_func_only = true
+filterwarnings = [
+  "error::temporalio.worker._workflow_instance.UnfinishedUpdateHandlerWarning",
+  "error::temporalio.worker._workflow_instance.UnfinishedSignalHandlerWarning",
+  "ignore::pytest.PytestDeprecationWarning",
+  "ignore::DeprecationWarning",
+]
```

Comment:
In order to convert the warnings to errors in the test suite, we have to re-enable warnings globally. I've ignored the deprecation warnings that I was able to but I've so far been unable to silence the following slightly annoying warnings:

```
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.MessageMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.ScalarMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.

.venv/lib/python3.12/site-packages/google/protobuf/internal/well_known_types.py:91
  /Users/dan/src/temporalio/sdk-python/.venv/lib/python3.12/site-packages/google/protobuf/internal/well_known_types.py:91: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    _EPOCH_DATETIME_NAIVE = datetime.datetime.utcfromtimestamp(0)
```

Reply:
I know we have a test to prove the warning happens, but I wonder if we need a test that proves that outside of pytest capture warnings, a dangling handler would raise.

I wasn't thinking about silencing warnings, I actually think they are fine. The main thing I wanted to make sure is that we failed tests that had dangling handlers unintentionally. And we can probably document for users that their pytest setups should also fail on these warnings.

Reply:
Yes, it would be nice to prove that the test suite is catching dangling handlers. Refactored the test to do that